### PR TITLE
DX - Refactor some of the sidebar frame function names to clarify behavior, and fix sidebar-sub-frame console error

### LINKED ISSUE #8471: Target error (benign?) in console for sidebar
## Describe the bug

- https://pixiebrix.slack.com/archives/C0436P48QHY/p1715865914941639

## To Reproduce

1. Activate the AA Copilot mapping demo mod
2. Open the mod in the sidebar
3. Check the console

## Screenshot

image

### PR DESCRIPTION
**Pull request by BLoe**
## What does this PR do?

- Closes #8471 
- Closely related to the discussion here with Federico & Todd about sidebar frames: https://github.com/pixiebrix/pixiebrix-extension/pull/7354#discussion_r1461563961
- Wraps the call to webext-messenger `getConnectedTarget()` in a try-catch to avoid console errors when it's called in a sub-frame in the sidebar
- Eventually we should implement communication between the top sidebar frame and the sub-frames so that they can also identify the currently connected browser tab to the sidebar

## Checklist

- [ ] Add jest or playwright tests and/or storybook stories
- [ ] Designate a primary reviewer

### CODE REVIEW

File: src/contrib/automationanywhere/aaFrameProtocol.ts [hunk 1]
```diff
@@ -121,14 +121,26 @@ export async function initCopilotMessenger(): Promise<void> {
...
-  setHostData(data);
+  try {
+    // Note: This code can be run either in the sidebar or in a modal. Currently,
+    // it sometimes also runs in nested frames in the MV3 sidebar, in which case
+    // the webext-messenger getTopLevelFrame() function currently cannot return
+    // a frameId/tabId, due to how it's implemented. It will throw an error, and
+    // in this specific case we don't want to be running the CoPilot Data
+    // initialization on a nested frame anyway, so we'll just eat the error and
+    // warn to console for now.
+    const connectedTarget = await getConnectedTarget();
```
Comment (twschiller) [inline src/contrib/automationanywhere/aaFrameProtocol.ts:132]: Just check before the call that this is coming from the top-level sidebar frame?

From a Dev Ex perspective, I'd at least recommend clarifying in the warning message that the warning is expected for sidebar subframes

File: src/contrib/automationanywhere/aaFrameProtocol.ts [hunk 2]
```diff
@@ -121,14 +121,26 @@ export async function initCopilotMessenger(): Promise<void> {
...
+    // Fetch the current data from the content script when the target frame loads
+    const data = await getCopilotHostData(connectedTarget);
+    console.debug("Setting initial Co-Pilot data", {
+      location: window.location.href,
+      data,
+    });
+    setHostData(data);
+  } catch (error) {
+    console.warn(
+      "Error getting connected target, aborting co-pilot messenger initialization",
```
Comment (twschiller) [inline src/contrib/automationanywhere/aaFrameProtocol.ts:142]: This error is also wrapping getCopilotHostData and setHostData, so the error message is misleading

**Review by twschiller — APPROVED** [review]
Conditionally approved - see comment on the scope of the try-catch and/or checking for top-frame status before making the call

File: src/contrib/automationanywhere/aaFrameProtocol.ts (hunk 1 above)
Comment (BLoe) [reply src/contrib/automationanywhere/aaFrameProtocol.ts:132]: I think this can also run in a modal, right? What check exactly are you talking about?

**Pull request merged**